### ISSUE TITLE
`git pull` breaks when remote not named "origin"

### ISSUE THREAD

**Issue by bencates** [git-lfs/git-lfs#2273]
I've found a few closed tickets that seem to be about the same issue (e.g. #948, #1015, #2105). They suggest renaming the remote to `origin` as a workaround, but no solution.

Any idea when this is going to actually be fixed? In the meantime, what are the odds of getting a better error message and maybe some mention of this limitation in the docs?

-----

```
➜  artoo git:(master) git pull
Updating ba827827..37f1f0c8
Downloading assets/about/hero.jpg (200.38 KB)
Error downloading object: assets/about/hero.jpg (87288e767bd09d4d38076e95a4d107d3fb679b1ad0cae7054ee21a5a760f0169)

Errors logged to /var/build/artoo/.git/lfs/objects/logs/20170526T135923.518976522.log
Use `git lfs logs last` to view the log.
error: external filter 'git-lfs filter-process' failed
fatal: assets/about/hero.jpg: smudge filter lfs failed
```

```
➜  artoo git:(master) ✗ git lfs logs last     
git-lfs/2.0.2 (GitHub; linux amd64; go 1.8)
git version 2.11.0

$ git-lfs filter-process
Error downloading object: assets/about/hero.jpg (87288e767bd09d4d38076e95a4d107d3fb679b1ad0cae7054ee21a5a760f0169)

Smudge error: Error downloading assets/about/hero.jpg (87288e767bd09d4d38076e95a4d107d3fb679b1ad0cae7054ee21a5a760f0169): batch response: Post /objects/batch: unsupported protocol scheme ""
github.com/git-lfs/git-lfs/errors.newWrappedError
	/tmp/docker_run/src/github.com/git-lfs/git-lfs/amd64/obj-x86_64-linux-gnu/src/github.com/git-lfs/git-lfs/errors/types.go:166
github.com/git-lfs/git-lfs/errors.NewSmudgeError
	/tmp/docker_run/src/github.com/git-lfs/git-lfs/amd64/obj-x86_64-linux-gnu/src/github.com/git-lfs/git-lfs/errors/types.go:252
github.com/git-lfs/git-lfs/lfs.PointerSmudge
	/tmp/docker_run/src/github.com/git-lfs/git-lfs/amd64/obj-x86_64-linux-gnu/src/github.com/git-lfs/git-lfs/lfs/pointer_smudge.go:68
github.com/git-lfs/git-lfs/lfs.(*Pointer).Smudge
	/tmp/docker_run/src/github.com/git-lfs/git-lfs/amd64/obj-x86_64-linux-gnu/src/github.com/git-lfs/git-lfs/lfs/pointer.go:64
github.com/git-lfs/git-lfs/commands.smudge
	/tmp/docker_run/src/github.com/git-lfs/git-lfs/amd64/obj-x86_64-linux-gnu/src/github.com/git-lfs/git-lfs/commands/command_smudge.go:63
github.com/git-lfs/git-lfs/commands.filterCommand
	/tmp/docker_run/src/github.com/git-lfs/git-lfs/amd64/obj-x86_64-linux-gnu/src/github.com/git-lfs/git-lfs/commands/command_filter_process.go:65
github.com/git-lfs/git-lfs/vendor/github.com/spf13/cobra.(*Command).execute
	/tmp/docker_run/src/github.com/git-lfs/git-lfs/amd64/obj-x86_64-linux-gnu/src/github.com/git-lfs/git-lfs/vendor/github.com/spf13/cobra/command.go:477
github.com/git-lfs/git-lfs/vendor/github.com/spf13/cobra.(*Command).Execute
	/tmp/docker_run/src/github.com/git-lfs/git-lfs/amd64/obj-x86_64-linux-gnu/src/github.com/git-lfs/git-lfs/vendor/github.com/spf13/cobra/command.go:551
github.com/git-lfs/git-lfs/commands.Run
	/tmp/docker_run/src/github.com/git-lfs/git-lfs/amd64/obj-x86_64-linux-gnu/src/github.com/git-lfs/git-lfs/commands/run.go:68
main.main
	/tmp/docker_run/src/github.com/git-lfs/git-lfs/amd64/obj-x86_64-linux-gnu/src/github.com/git-lfs/git-lfs/git-lfs.go:35
runtime.main
	/usr/local/go/src/runtime/proc.go:185
runtime.goexit
	/usr/local/go/src/runtime/asm_amd64.s:2197

ENV:
LocalWorkingDir=/var/build/artoo
LocalGitDir=/var/build/artoo/.git
LocalGitStorageDir=/var/build/artoo/.git
LocalMediaDir=/var/build/artoo/.git/lfs/objects
LocalReferenceDir=
TempDir=/var/build/artoo/.git/lfs/tmp
ConcurrentTransfers=3
TusTransfers=false
BasicTransfersOnly=false
SkipDownloadErrors=false
FetchRecentAlways=false
FetchRecentRefsDays=7
FetchRecentCommitsDays=0
FetchRecentRefsIncludeRemotes=true
PruneOffsetDays=3
PruneVerifyRemoteAlways=false
PruneRemoteName=origin
AccessDownload=none
AccessUpload=none
DownloadTransfers=basic
UploadTransfers=basic
GIT_DIR=.git
GIT_PREFIX=
GIT_REFLOG_ACTION=pull
```

**Comment by ttaylorr** [comment]
@bencates thanks so much for opening this and linking to a few other related issues. In scanning through open LFS issues, I found https://github.com/git-lfs/git-lfs/issues/2169, where I left https://github.com/git-lfs/git-lfs/issues/2169#issuecomment-296695991, which is how I think this fix should be implemented.

If it's alright with you, I'm going to move discussion over there. I've milestoned that issues into the 2.2.0 release, so it should be fixed then. Thanks again!
